### ISSUE TITLE
Documentation updates for 1.0.0-beta4

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j-docs#174]
Here's a summary list of features and changes in beta4 that the docs will need to be updated for:

**ND4J**
* ND4J DataTypes support/changes
* Nd4j. new methods: createFromArray, DataType overloads for existing methods
* Update memory management page (periodic GC disabled by default; INDArray.close() method)
* Check pages for rank 0/1 support/changes
* Review, update and consolidate old pages
* Numpy NPZ support
* Reduction axis change (in op not as exec arg)

**DL4J**
* DL4J: MKL-DNN support
* Notes on DataTypes (datatype config) and datatype conversion.
* Update quickref guide: new layers (attention etc). L1/L2/Weight decay - classes now. Weight init: classes now. GELU activation. BERT iterator. Embedding layer word vectors based initialization. Capsule network layers. Cifar10DataSetIterator. Wassrstein loss.
* Evaluation: update for rank 4 support, and axis argument
* Check docs on ParallelInference; add notes on ModelAdapter
* NLP pages: add references/links for BERT WordPiece tokenizers. Notes on next format serialization
* UI page - add section on multi-user/multi-session support
* Update CUDA page for new CUDA versions

**DataVec**
* PythonTransform
* Add new transforms: FirstDigitTransform, TokenizerBagOfWordsTermSequenceIndexTransform, GazeteerTransform
* StreamInputSplit

**SameDiff**
* SameDiff docs in general
    - Tensorflow import - examples, what is/isn't supported, etc
* TensorFlowImportValidator, GraphTransformUtil, convertToVariable/convertToConstant
* SameDiff UI

**Arbiter**
* Genetic algorithm search
